norduser/utmp.go

```diff
@@ -0,0 +1,99 @@
+package norduser
+
+/*
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <utmpx.h>
+
+const int ERROR_REALLOC = -1;
+const int ERROR_MALLOC_USERNAME = -2;
+
+void free_users_table(char*** users, int size) {
+  for (int index = 0; index < size; index++) {
+    if ((*users)[index] != NULL) {
+      free((*users)[index]);
+      (*users)[index] = NULL;
+    }
+  }
+
+  if (*users != NULL) {
+    free(*users);
+    *users = NULL;
+  }
+}
+
+int get_utmp_user_processes(char*** users) {
+  int index = 0;
+  int size = 0;
+  setutxent();
+
+  *users = NULL;
+
+  struct utmpx* u;
+  while ((u = getutxent()) != NULL) {
+    if (u->ut_type != USER_PROCESS) {
+      continue;
+    }
+
+    if (index == size) {
+      char** tmp;
+      tmp = realloc(*users, (size + 1) * sizeof(char*));
+      if (tmp == NULL) {
+        free_users_table(users, index);
+        endutxent();
+        return ERROR_REALLOC;
+      }
+      *users = tmp;
+
+      size++;
+    }
+
+    (*users)[index] = malloc(__UT_NAMESIZE + 1);
+    (*users)[index][__UT_NAMESIZE]='\0';
```

Comment:
better move this after the if, otherwise might crash if `(*users)[index] == NULL)`